[CHORE] Remove react-bootstrap

c-py opened the pull request:
# Description

Removes `react-bootstrap` dependency. Moves `Modal` over to HeadlessUI which is used in the Megastore.

## Changes

- Renamed imports of `react-bootstrap` to `ui/components/Modal`
- - Removed support of `centered` attribute
- - Removed support of `keyboard` attribute
- - Removed support of `Modal.Header`, `Modal.Body` and `Modal.Footer`

- Removed import of `Carousel`


## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

localhost

# Checklist:

- [X] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [ ] I have read the contributing guidelines and agree to the T&Cs
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

c-py commented on src/components/ui/Modal.tsx:
```diff
@@ -0,0 +1,85 @@
+    <Transition appear show={show} as={Fragment}>
+      <Dialog
+        as="div"
+        className="fixed inset-0 flex min-h-full items-center justify-center z-50"
```
Bootstrap was sitting at 1050, now Modal is sitting at `z-50` just above the Hud at `z-40`. Anima/Synaps sits at 1000.

adamhannigan reviewed (commented):
Nice work mate! Code LGTM 🚀 

Here are some UI issues I came across:


1. Running the app in art mode, I receive a cannot read context error (see attached)
2. When loading, no 'Loading' panel appears. I just get a blank blue ocean
3. A strange play button is covering the "Save" Icon in the bottom right (see attached)
4. Scrolling not working 
5. The icon for Home in the travel menu has lost it's padding and overlaps
6. When I got a server error, I was unable to click the "Refresh" button in the modal. I had to refresh the page to continue playing
7. When buying block bucks and confirming, the content in the panel just disappears and I cannot continue. It feels like we are missing a loading state for it

<img width="407" alt="Screenshot 2024-02-19 at 2 16 36 pm" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/7cc95627-0c65-41ec-b1f0-89cdaedb3684">
<img width="600" alt="Screenshot 2024-02-19 at 2 16 55 pm" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/448035c3-8709-481d-8972-7e27ca4dcca3">

<img width="163" alt="Screenshot 2024-02-19 at 2 17 58 pm" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/6ca6d32e-1423-47ba-95b8-f82c811d9e45">
<img width="173" alt="Screenshot 2024-02-19 at 2 19 46 pm" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/6db014b9-0a4d-4473-b496-0a8d1b422a4e">

c-py commented:
Nice finds @adamhannigan. I think this came about as I removed bootstraps CSS as the last thing in the PR. I have addresses the comments:

> Running the app in art mode, I receive a cannot read context error (see attached)

I was missing some optional chaining from the settings menu after updating the `Share` screen 😬.


> When loading the game, no 'Loading' panel appears. I just get a blank blue ocean

Seems to be a bit of a z-index kerfuffle. The ocean was rendering at `1050`, the same z-index as bootstrap modal. I lowered this to 49 to be underneath the z-50 headlessUI modal.

> A strange play button is covering the "Save" Icon in the bottom right (see attached)

The `audio` element used to play game music was using a bootstrap class `d-none`.

> Scrolling not working

Bootstrap was forcing overflow scroll as !important. I have updated the `ScrollContainers` to do the same.

> The icon for Home in the travel menu has lost it's padding and overlaps

Bootstrap was enforcing py-2 as !important. I have updated the island list to do the same.


> When I got a server error, I was unable to click the "Refresh" button in the modal. I had to refresh the page to continue playing

Something adds `pointer-events: none` to the `body` when the error boundary is hit. I think it may be react. I have updated the modal to reset this with `pointer-events:auto`


> When buying block bucks and confirming, the content in the panel just disappears and I cannot continue. It feels like we are missing a loading state for it

To be fixed in another PR

adamhannigan reviewed (approved):
LGTM ✅ All working well now
